### PR TITLE
Replace DEBUG with EXIV2_DEBUG_MESSAGES

### LINKED ISSUE #926: Please change the behaviour of debug stdout/stderr output.
I have a kind request about the exiv2 sources. I see that debug builds use the define `DEBUG` to print more verbose output messages. These messages are directly written to `std::cout` or `std::cerr`.

This behavior is quite common and there is really nothing wrong with it. However I would slightly prefer more control over the messages, because when I use exiv2 debug builds, they tend to be quite verbose on stdout/stderr.

Would it be acceptable to change the logging behavior? I could think to either add a new define like `EXIV2_DEBUG_MESSAGES` that users can separately control independent of `DEBUG`. Or to leave the messages to be printed only when `DEBUG` is defined, but print via exiv2's existing logging mechanism from `error.hpp` so users can control the verbosity at runtime.

Does any of this sound reasonable? If you find it not worthwhile, please feel free to close this kind request.

### PR DESCRIPTION
**Pull request by piponazo**
Fixes #926

Same than #938 but for master. Note that we might end up not merging this change into 0.27-maintenance

### CODE REVIEW

File: src/pngimage.cpp [hunk 1]
```diff
@@ -59,7 +59,7 @@ namespace
         0xe7, 0x00, 0x00, 0x00, 0x00, 0x49, 0x45, 0x4e, 0x44, 0xae, 0x42, 0x60, 0x82};
     inline bool compare(const char* str, const Exiv2::DataBuf& buf, size_t length)
     {
-        // str & length should compile time constants => only running this in DEBUG mode is ok
+        // str & length should compile time constants => only running this in EXIV2_DEBUG_MESSAGES mode is ok
```
Comment (D4N) [inline src/pngimage.cpp]: This comment is wrong now.

File: README.md [hunk 2]
```diff
@@ -425,12 +425,12 @@ $
...
 [TOC](#TOC)
 
-2) About preprocessor symbols **NDEBUG** and **DEBUG**
+2) About preprocessor symbols **NDEBUG** and **EXIV2_DEBUG_MESSAGES**
 
 In accordance with current practice, exiv2 respects the symbol *NDEBUG* which is set for Release builds.  There are sequences of code which are defined within:
 
 ```C++
-#ifdef DEBUG
+#ifdef EXIV2_DEBUG_MESSAGESDEBUG
```
Comment (D4N) [inline README.md]: The preprocessor symbol name is wrong and it doesn't make sense given the next paragraph which statest that this is "not compiled for debug builds".

This paragraph should imho instead explain that this preprocessor symbol is for debug messages.

**Review by D4N — CHANGES_REQUESTED** [review]
LGTM, I've flagged two small inconsistencies in the docs.

**Comment by piponazo** [comment]
@D4N I have addressed the two comments you've made. Let me know if you are happy with the new version 😉

**Review by D4N — APPROVED** [review]

**Comment by emmenlau** [comment]
Awesome, thanks!

**Pull request merged**